tools/osx-notarize.sh

```diff
@@ -1,34 +1,67 @@
 #!/bin/sh
 
-# Uses gon, from https://github.com/mitchellh/gon, to notarize a generated node-<version>.pkg file
-# with Apple for installation on macOS Catalina and later as validated by Gatekeeper.
+# Notarize a generated node-<version>.pkg file as an Apple requirement for installation on macOS Catalina and later, as validated by Gatekeeper.
+# Uses gon (Xcode version < 13.0) or notarytool (Xcode >= 13.0).
 
 set -e
 
-gon_version="0.2.2"
-gon_exe="${HOME}/.gon/gon_${gon_version}"
-
+xcode_version=$(xcodebuild -version | awk '/Xcode/ {print $2}')
 pkgid="$1"
 
-[ -z "$pkgid" ] && \
-  echo "Usage: $0 <pkgid>" \
+if [ -z "$pkgid" ]; then
+  echo "Usage: $0 <pkgid>"
   exit 1
+fi
 
 # shellcheck disable=SC2154
-[ -z "$NOTARIZATION_ID" ] && \
-  echo "No NOTARIZATION_ID environment var. Skipping notarization." \
+if [ -z "$NOTARIZATION_ID" ]; then
+  echo "No NOTARIZATION_ID environment variable. Skipping notarization."
   exit 0
+fi
 
-set -x
+if [ "$(echo "$xcode_version < 13.0" | bc)" -eq 1 ]; then
+  echo "Notarization process is done with gon."
+  set -x
 
-mkdir -p "${HOME}/.gon/"
+  gon_version="0.2.2"
+  gon_exe="${HOME}/.gon/gon_${gon_version}"
 
-if [ ! -f "${gon_exe}" ]; then
-  curl -sL "https://github.com/mitchellh/gon/releases/download/v${gon_version}/gon_${gon_version}_macos.zip" -o "${gon_exe}.zip"
-  (cd "${HOME}/.gon/" && rm -f gon && unzip "${gon_exe}.zip" && mv gon "${gon_exe}")
-fi
+  mkdir -p "${HOME}/.gon/"
+
+  if [ ! -f "${gon_exe}" ]; then
+    curl -sL "https://github.com/mitchellh/gon/releases/download/v${gon_version}/gon_${gon_version}_macos.zip" -o "${gon_exe}.zip"
+    (cd "${HOME}/.gon/" && rm -f gon && unzip "${gon_exe}.zip" && mv gon "${gon_exe}")
+  fi
 
-sed -e "s/{{appleid}}/${NOTARIZATION_ID}/" -e "s/{{pkgid}}/${pkgid}/" tools/osx-gon-config.json.tmpl \
-  > gon-config.json
+  sed -e "s/{{appleid}}/${NOTARIZATION_ID}/" -e "s/{{pkgid}}/${pkgid}/" tools/osx-gon-config.json.tmpl \
+    > gon-config.json
 
-"${gon_exe}" -log-level=info gon-config.json
+  "${gon_exe}" -log-level=info gon-config.json
+
+else
+  echo "Notarization process is done with Notarytool."
+
+  if ! command -v xcrun >/dev/null || ! xcrun --find notarytool >/dev/null; then
+    echo "Notarytool is not present in the system. Notarization has failed."
+    exit 1
+  fi
+
+  # Submit the package for notarization
+  notarization_output=$(
+    xcrun notarytool submit "node-$pkgid.pkg" \
+      --apple-id "@env:NOTARIZATION_APPLE_ID" \
+      --password "@env:NOTARIZATION_PASSWORD" \
+      --team-id "@env:NOTARIZATION_TEAM_ID" \
+      --wait 2>&1
```

Comment:
Great job 👏🏼 

If I could share an idea, I would say that storing its credentials with `notarytool store-credentials` and then using `--keychain-profile` option for the `notarytool submit` is probably a safer way.

You can handle the credential locally or on the CI and expose the `--keychain-profile` within the code (the `tools/osx-notarize.sh` script in that case).

Not a blocker, I would say, but definitely an improvement :)

Reply:
```suggestion
    xcrun notarytool submit "node-$pkgid.pkg" \
      --keychain-profile "@env:KEYCHAIN_PROFILE" \
      --wait 2>&1
```

Reply:
Thanks @tony-go! I also agree that `--keychain-profile` is a safer way, but we will need to see how/when we can introduce the store-credentials step (maybe when ansible the vms?), as we will need to have the profile created before using it.

Just for [reference](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow):

```
xcrun notarytool store-credentials "AC_PASSWORD"
               --apple-id "AC_USERNAME"
               --team-id <WWDRTeamID>
               --password <secret_2FA_password>
```

Reply:
Yeah, I already used it :)

Let me know If I can help in the future ^^

Thanks again

Reply:
Hopefully `notarytool` doesn't have whatever issue we had with `altool` and retrieving stuff out of the keychain: https://github.com/nodejs/build/pull/2199#issuecomment-593831495